pkg/otlp/collector.go

```diff
@@ -153,20 +157,32 @@ func (p *Pipeline) Stop() {
 func BuildAndStart(ctx context.Context, cfg config.Config, s serializer.MetricSerializer) (*Pipeline, error) {
 	pcfg, err := FromAgentConfig(config.Datadog)
 	if err != nil {
-		return nil, fmt.Errorf("config error: %w", err)
+		pipelineError = fmt.Errorf("config error: %w", err)
+		return nil, pipelineError
 	}
 
 	p, err := NewPipeline(pcfg, s)
 	if err != nil {
-		return nil, fmt.Errorf("failed to build pipeline: %w", err)
+		pipelineError = fmt.Errorf("failed to build pipeline: %w", err)
+		return nil, pipelineError
 	}
 
 	go func() {
 		err = p.Run(ctx)
 		if err != nil {
-			log.Errorf("Error running the OTLP pipeline: %s", err)
+			pipelineError = fmt.Errorf("Error running the OTLP pipeline: %w", err)
+			log.Errorf(pipelineError.Error())
 		}
 	}()
 
 	return p, nil
 }
+
+// GetCollectorStatus get the collector status and error message (if there is one)
+func GetCollectorStatus(p *Pipeline) (string, string) {
```

Comment:
I think this would be easier to understand if you either returned a `Status` struct with fields for each of these, or if you used [named return values](https://go.dev/tour/basics/7) here. The struct solution may be better in the long term since it's likely that what we report as a status will change in the future, but either works for now!

Reply:
Fixed with [c00d7b9](https://github.com/DataDog/datadog-agent/pull/11822/commits/c00d7b92c3a4d11eb98563e28ac35b0d65b5362e)